build.gradle

```diff
@@ -142,223 +159,209 @@ task cucumberTestReport(type: TestReport) {
-    testCompile "com.mattbertolini:liquibase-slf4j"
-    testCompile "org.hamcrest:hamcrest-library"
-    testCompile "com.h2database:h2"
-    testCompile(group: 'org.testcontainers', name: 'testcontainers', version: '1.10.5') {
-        exclude group: 'commons-io', module: 'commons-io'
-    }
-    testCompile group: 'org.testcontainers', name: 'postgresql', version: '1.10.5'
-
-    testCompile("com.github.database-rider:rider-spring:1.2.7") {
-        exclude module: 'slf4j-api'
-        exclude module: 'slf4j-simple'
+    testImplementation "com.jayway.jsonpath:json-path"
+    testImplementation "io.cucumber:cucumber-junit"
+    testImplementation "io.cucumber:cucumber-spring"
+    testImplementation ("org.springframework.boot:spring-boot-starter-test") {
+        exclude module: "junit"
+        exclude group: "com.vaadin.external.google", module: "android-json"
     }
+    testImplementation "org.springframework.security:spring-security-test"
+    testImplementation "org.springframework.boot:spring-boot-test"
+    testImplementation "org.junit.jupiter:junit-jupiter-engine"
+    testImplementation "org.junit.vintage:junit-vintage-engine"
+    testImplementation "com.tngtech.archunit:archunit-junit5-api:${archunit_junit5_version}"
+    testRuntimeOnly "com.tngtech.archunit:archunit-junit5-engine:${archunit_junit5_version}"
+    testImplementation "org.assertj:assertj-core"
+    testImplementation "junit:junit"
+    testImplementation "org.mockito:mockito-core"
+    testImplementation "org.hamcrest:hamcrest-library"
+    testImplementation "com.h2database:h2"
     liquibaseRuntime "com.h2database:h2"
-    testCompileOnly "org.projectlombok:lombok:${lombok_version}"
-
     //jhipster-needle-gradle-dependency - JHipster will add additional dependencies here
 
-    compileOnly "org.projectlombok:lombok:${lombok_version}"
-    compile "com.icthh.xm.commons:xm-commons-timeline:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-security:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-migration-db:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-tenant-endpoint:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-permission:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-ms-web:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-lep:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-i18n:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-metric:${xm_commons_version}"
-    compile "com.icthh.xm.commons:xm-commons-logging-web:${xm_commons_version}"
-}
-
-checkstyle {
-    toolVersion = "7.6.1"
-    ignoreFailures = true
-    configFile = rootProject.file('./qa/checkstyle.xml')
-}
+    // XM custom dependencies
+    implementation "org.hibernate:hibernate-envers"
+    implementation 'com.vladmihalcea:hibernate-types-52:2.4.3'
+    implementation 'com.vladmihalcea:hibernate-types-52:2.4.3'
```

Comment:
why this dependencie duplicated? And what purpose of this library?

Reply:
This library is need for successful compilation: 
eg. `com.vladmihalcea.hibernate.type.json.JsonBinaryType;` in `JsonbTypeRegistrator.java:7`